src/coreclr/jit/compiler.hpp

```diff
@@ -966,28 +1065,31 @@ inline regNumber genFirstRegNumFromMaskAndToggle(regMaskTP& mask)
 
     regNumber regNum = (regNumber)BitScanForward(mask);
 
-    mask ^= genRegMask(regNum);
+    mask ^= regNum;
```

Comment:
Can we have a `RemoveRegNumFromMask` for `SingleTypeRegSet`? It seems odd to me to xor a bit mask with an index like this.